### PR TITLE
feat: support Node 10

### PR DESCRIPTION
**Pull request by DominicKramer**

### CODE REVIEW

**Comment by DominicKramer** [comment]
The tests I have can be used now.  Another option I am looking into is adding testing against the Node.js master branch.

**Comment by DominicKramer** [comment]
PTAL

File: test/test-v8debugapi.ts [hunk 1]
```diff
@@ -145,9 +183,10 @@ describe('debugapi selection', () => {
               assert.ok(api instanceof v8debugapi.V8DebugApi);
             }
             if (process.env.GCLOUD_USE_INSPECTOR &&
-                semver.satisfies(process.version, '<8')) {
+                utils.satisfies(process.version, '<8')) {
               assert(logText.includes(utils.messages.INSPECTOR_NOT_AVAILABLE));
             } else {
+              console.log(`logText='${logText}'`);
```
Comment (kjin) [inline test/test-v8debugapi.ts]: Nit: Remove debug line

File: test/test-v8debugapi.ts [hunk 2]
```diff
@@ -156,6 +195,38 @@ describe('debugapi selection', () => {
   });
 });
 
+const describeFn = utils.satisfies(process.version, '>=10') ? describe : describe.skip;
+describeFn('debugapi selection on Node >=10', () => {
+  const config: ResolvedDebugAgentConfig = extend(
+      {}, defaultConfig, {workingDirectory: __dirname, forceNewAgent_: true});
+  const logger =
+      new common.logger({levelLevel: config.logLevel} as {} as LoggerOptions);
```
Comment (kjin) [inline test/test-v8debugapi.ts]: Typo?

File: test/test-v8debugapi.ts (hunk 1 above)
Comment (DominicKramer) [reply test/test-v8debugapi.ts]: Fixed

File: test/test-v8debugapi.ts (hunk 2 above)
Comment (DominicKramer) [reply test/test-v8debugapi.ts]: I don't think this is a typo.

**Pull request merged**